superset-frontend/src/views/CRUD/data/database/types.ts

```diff
@@ -52,3 +54,51 @@ export type DatabaseObject = {
   allow_csv_upload?: boolean;
   extra?: string;
 };
+
+export type DatabaseForm = {
+  engine: string;
+  name: string;
+  parameters: {
+    properties: {
+      database: {
+        description: string;
+        type: string;
+      };
+      host: {
+        description: string;
+        type: string;
+      };
+      password: {
+        description: string;
+        nullable: boolean;
+        type: string;
+      };
+      port: {
+        description: string;
+        format: string;
+        type: string;
+      };
+      query: {
+        additionalProperties: {};
+        description: string;
+        type: string;
+      };
+      username: {
+        description: string;
+        nullable: boolean;
+        type: string;
+      };
+    };
+    required: string[];
+    type: string;
+  };
+  preferred: boolean;
+  sqlalchemy_uri_placeholder: string;
+};
+
+// the values should align with the database
+// model enum in superset/superset/models/core.py
+export enum CONFIGURATION_METHOD {
+  SQLALCHEMY_URI = 'sqlalchemy_form',
```

Comment:
This was changed to sqlalchemy_form in the backend:
https://github.com/preset-io/superset/blob/master/superset/models/core.py#L103

Reply:
it shouldn't matter what the key is on the front end, but yeah, maybe for consistency I'll change it in a future iteration. Thanks for pointing that out!

Reply:
yup! I meant for just consistency.

Reply:
We should also figure out a way to centralize this enum across the backend and frontend